logic/src/main/java/nl/tehdreamteam/se42/logic/validator/Validator.java

```diff
@@ -0,0 +1,55 @@
+package nl.tehdreamteam.se42.logic.validator;
+
+import nl.tehdreamteam.se42.logic.exception.ServerError;
+
+/**
+ * A {@code Validator} can validate a certain object and return a {@link ServerError} when a validation fails.
+ * The {@code Validator} class provides a common abstract platform for the implementation of {@code Validator}.
+ *
+ * @author Oscar de Leeuw
+ */
+public abstract class Validator implements Comparable<Validator> {
+
+    private final ServerError error;
+    private final int priority;
+
+    protected Validator(ServerError error, int priority) {
+        this.error = error;
+        this.priority = priority;
+    }
+
+    protected Validator(ServerError error, Priority priority) {
+        this(error, priority.value());
+    }
+
+    public ServerError getError() {
+        return error;
+    }
+
+    @Override
+    public int compareTo(Validator o) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Override "equals(Object obj)" to comply with the contract of the "compareTo(T o)" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonarqube.tehmessiah.nl/coding_rules#rule_key=squid%3AS1210)